### DISCUSSION TITLE
LazyVim and Textwidth

### DISCUSSION THREAD

**Discussion by m4kman** [LazyVim/LazyVim#2978] [Q&A]
Hi,

I'm fairly new to using LazyVim (recently switched from VS Code) and I've been really liking it so far. Faced various caveats on the way which I managed to solve, except one.

I can't for the life of me figure out **how to force textwidth** in LazyVim. I've tried using the set command, putting it in the options.lua file, appending the formatoptions with t (even though it comes in the full-spec).

All of this to no avail (yes, wrap is enabled). The **gq** keymap does not work either, and I'm pretty much lost. I have tried searching but couldn't resolve it which led to this post.

For insight, I'll attach a screenshot so you can visualize the issue. An example use-case would be the notorious long lines of classes while using Tailwinds.

![image](https://github.com/LazyVim/LazyVim/assets/30337413/144ff2cb-86b4-4a28-a789-d540c104af21)

As you can see, the classes string is taking the whole line instead of respecting the max textwidth and wrapping. I'm using **Prettier** as my formatter, and I've even tried configuring it with the printWidth option. 

 Any help in this will be really appreciated. Please feel free to provide any other suggestions/recommendations you might have. Thanks.

**Comment by dpetka2001** [comment]
Can you try in your `prettier` config file to set `"proseWrap": "always"` and see if that makes a difference?

**Reply by m4kman** [reply]
Thanks for the reply. 

I believe proseWrap is useful when dealing with markdown files so it wouldn't be helpful here. Nonetheless, I tried it and the problem remains.

**Reply by dpetka2001** [reply]
There is also a [prettier plugin](https://www.npmjs.com/package/prettier-plugin-classnames) that seems to be doing this. Maybe you would like to check it out. Otherwise, I found [this discussion](https://github.com/tailwindlabs/tailwindcss/discussions/7763) that seems to be wanted as a feature for tailwind classes, but doesn't seem to be implemented yet? Or maybe the `prettier-tailwindcss-plugin` has implemented that? Not sure.

**Reply by m4kman** [reply]
Great find, thank you. This should help me with Tailwind classes. 

I would like to resolve the textwidth issue in general, as well, so that my LazyVim config can respect the maximum characters per line and wrap automatically upon write in any case.

**Reply by dpetka2001** [reply]
Wrapping automatically on save is purely dependant on the formatter you use.

**Reply by m4kman** [reply]
Yes, which is why I've opened this discussion for any help in forcing the formatter to respect the textwidth option.

**Reply by dpetka2001** [reply]
This is a problem with the formatter. Specifically `prettier` doesn't seem to break strings. There's an issue [here](https://github.com/prettier/prettier/issues/4123) that got closed as not going to be implemented in core, rather it would be better done in a plugin. It really depends on what each formatter supports. You'd be better creating a discussion in the formatter repo you're interested in and maybe request as a feature. There really is not much that can be done from LazyVim's side if a formatter doesn't format in the way you would like it to.

**Reply by dpetka2001** [reply]
You could also use `gw` to format over the lines that you want and then use formatter specific ignore indicators for the lines you don't want formatting. So, for prettier, you could use `gw` over some lines and then before those lines put a `<!-- prettier-ignore -->`, so that they don't get formatted by prettier upon saving the file.

**Reply by samuelhoover** [reply]
@dpetka2001 Thank you for mentioning `gw`, that was exactly what I was looking for since I prefer hard wrapping when writing in LaTeX. I couldn't find that keymap in the available list, how can I find the full list of keymaps

**Reply by dpetka2001** [reply]
This is a builtin Neovim keymap. On the LazyVim website only the keymaps that are configured by LazyVim are mentioned, not the Neovim builtins. For more info see `:h formatting`.

**Reply by samuelhoover** [reply]
Thanks for the info! I used to use `gq` for hard wrapping, but when I switched to LazyVim that keymap stopped working. I'll check Neovim keymaps next time I can't seem to find it through LazyVim. Thanks again!

**Reply by dpetka2001** [reply]
`gq` takes into consideration if `formatexpr` is set and LazyVim sets that, so it can use either `conform.nvim` when it's available or the LSP formatting as fallback. `gw` formats without taking `formatexpr` into account, hence what you're observing.
